### PR TITLE
Protocol refactor

### PR DESCRIPTION
**Pull request by sinui0**
This PR refactors our "Protocol" section to sync with our newer communication strategy. The section is not complete but this is a start.

I moved the "Commitment" and "Signing" sections into 1 section under "Notarization".

### CODE REVIEW

File: src/png-diagrams/mpc-decryption.png
Comment (heeckhau) [inline src/png-diagrams/mpc-decryption.png:1]: the `drawio` file is missing for the `mpc-decryption`

File: src/protocol/notarization.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# Notarization
+
+As part of the TLSNotary protocol, the `Prover` can create authenticated commitments to the plaintext and have the `Notary` sign them without ever seeing the plaintext. This offers a way for the `Prover` to selectively prove the authenticity of arbitrary portions of the plaintext to a different `Verifier`.
```
Comment (heeckhau) [inline src/protocol/notarization.md:3]: We will have to introduce the notary here again, as a special case where.

I may also be better to move this chapter to the end of the protocol chapters.

**Review by heeckhau — CHANGES_REQUESTED** [review]
I think I am reviewing this in the wrong order. I'll review the introduction first, and next see how this PR fits in the flow.

**Review by themighty1 — APPROVED** [review]

File: src/diagrams/key_exchange.svg
Comment (themighty1) [inline src/diagrams/key_exchange.svg:1]: requester is still present in the new diagram

File: src/glossary.md [hunk 2]
```diff
@@ -2,21 +2,23 @@
 
 | Term  | Explanation                                     |
 | ----- | ----------------------------------------------- |
-| AES   | Advanced Encryption Standard                    |
+| 2PC   | Two-party computation                           |
```
Comment (themighty1) [inline src/glossary.md:5]: Secure two-party computation

File: src/glossary.md [hunk 3]
```diff
@@ -2,21 +2,23 @@
...
 | ECB   | Electronic codebook (encryption mode)           |
+| ECDH  | Elliptic-Curve Diffie-Hellman                   |
 | GC    | Garbled Circuit                                 |
 | GCM   | Galois/Counter Mode                             |
 | GHASH | GCM hash                                        |
 | HMAC  | Hash-based Message Authentication Code          |
-| MAC   | Message Authentication Code                     |
 | M2a   | Multiplication-to-Addition                      |
+| MAC   | Message Authentication Code                     |
+| MPC   | Multi-party computation                         |
```
Comment (themighty1) [inline src/glossary.md:17]: secure multi-party computation

File: src/protocol/mpc-tls/README.md [hunk 4]
```diff
@@ -1,9 +1,21 @@
...
 
 Listed below are some key points regarding this process:
 
- - The identity of the `Server` can be hidden from the `Verifier`, while the `Prover` is still capable of proving the `Server` identity later.
  - The `Verifier` only ever sees the *encrypted* application data of the TLS session.
  - The protocol guarantees that the `Prover` is not solely capable of constructing requests, nor can they forge responses from the `Server`.
+
+When TLSNotary is used in a setup with a general-purpose `Verifier` (a.k.a a `Notary`) to verify the TLS session, the identity of the `Server` can be hidden from this `Verifier`. This makes the data portable to other, application-specific verifiers.
+
+The MPC-TLS consists of following steps:
```
Comment (themighty1) [inline src/protocol/mpc-tls/README.md:12]: the following steps

File: src/protocol/mpc-tls/README.md [hunk 5]
```diff
@@ -1,9 +1,21 @@
...
+
+1. **Handshake**  
+A TLS handshake is the first step in establishing a TLS connection between the `Prover`/`Verifier` and the `Server`. The result of this handshake is a *Pre Master Secret (PMS)*, a symmetrical key that will be used for further encrypted communication. The server has the full key; the `Prover` and the `Verifier` only have their share of this key.
```
Comment (themighty1) [inline src/protocol/mpc-tls/README.md:15]: "key that will be used for encrypt further communication"

**Review by themighty1 — APPROVED** [review]
lgtm, sry github's "suggest a  change" didnt work for some reason

File: src/protocol/mpc-tls/README.md (hunk 5 above)
Comment (themighty1) [reply src/protocol/mpc-tls/README.md:15]: sry. misread the original text, ignore this comment.

**Review by heeckhau — APPROVED** [review]

File: src/protocol/mpc-tls/README.md [hunk 6]
```diff
@@ -1,9 +1,21 @@
...
+
+When TLSNotary is used in a setup with a general-purpose `Verifier` (a.k.a a `Notary`) to verify the TLS session, the identity of the `Server` can be hidden from this `Verifier`. This makes the data portable to other, application-specific verifiers.
+
+The MPC-TLS consists of the following steps:
+
+1. **Handshake**  
+A TLS handshake is the first step in establishing a TLS connection between the `Prover`/`Verifier` and the `Server`. The result of this handshake is a *Pre Master Secret (PMS)*, a symmetrical key that will be used for further encrypted communication. The server has the full key; the `Prover` and the `Verifier` only have their share of this key.
+2. **Encryption, Decryption, and MAC Computation**  
+Next, the `Prover` and `Verifier` use MPC to encrypt, and decrypt, data sent to, and received from, the `Server`. They also compute a *Message Authentication Code (MAC)* 
+for the data that ensures untampered communication.
```
[the remark this answers is not in the document]
Comment (heeckhau) [reply src/protocol/mpc-tls/README.md:18]: OK. I wanted to give a short summary of what is coming next here.
I'll put it in comments for now.

**Pull request merged**